### PR TITLE
Add RF range sensor

### PR DESCRIPTION
**Pull request by caguero**
This pull request adds a new RF range sensor that can be attached to a sensor slot. The sensor periodically publishes an `ignition::msgs::Param_V` message with the following fields:

* `model`: Unscoped name of another model with an RF range sensor attached.
* `range`: Range in meters to the detected RF range sensor.
* `rssi`: Signal strength received from the detected RF range sensor.

**How to test it.**

1. Launch an environment:
```
ros2 launch mbzirc_ros competition_local.launch.py ign_args:="-v 4 -r simple_demo.sdf"
```

2. Launch `quadrotor_1`:
```
ros2 launch mbzirc_ign spawn.launch.py name:=quadrotor_1 world:=simple_demo model:=mbzirc_quadrotor type:=uav x:=-9.5 y:=2 z:=1.05 R:=0 P:=0 Y:=0 slot0:=mbzirc_hd_camera slot1:=mbzirc_rf_range
```

3. Launch `quadrotor_2`:
```
ros2 launch mbzirc_ign spawn.launch.py name:=quadrotor_2 world:=simple_demo model:=mbzirc_quadrotor type:=uav x:=9.5 y:=2 z:=1.05 R:=0 P:=0 Y:=0 slot0:=mbzirc_hd_camera slot1:=mbzirc_rf_range
```

4. Subscribe to one of the sensor topics, e.g. from `quadrotor_1`:
```
ign topic -e -t /world/simple_demo/model/quadrotor_1/model/sensor_1/rfsensor
```

You should observe something similar to this output running at 1Hz:

```
header {
  stamp {
    sec: 11
  }
}
param {
  params {
    key: "model"
    value {
      type: STRING
      string_value: "quadrotor_1"
    }
  }
  params {
    key: "range"
    value {
      type: DOUBLE
      double_value: 12.384630932840144
    }
  }
  params {
    key: "rssi"
    value {
      type: DOUBLE
      double_value: -43.414959792914587
    }
  }
}
```

Note that there's noise in the measurements. The noise is controlled by the `sigma` parameter in the plugin. If you change that parameter to `0.0`, you should get perfect measurements.

### CODE REVIEW

**Comment by iche033** [comment]
> We're looking to provide multiple configurations, e.g. one short range and one long range. Teams must use the same model on all robots. One idea is to let the sensor set one of the RF params that affects its working range when it's loaded, e.g. tx_power?

Did the changes in a separate PR: https://github.com/osrf/mbzirc/pull/155

**Comment by mjcarroll** [comment]
> This pull request adds a new RF range sensor that can be attached to a sensor slot. The sensor periodically publishes an ignition::msgs::Param_V message with the following fields:

Is `ignition::msgs::Param_V` necessary or is `ignition::msgs::Param` sufficient?  It appears that we are only adding a single "Param" entry in the Param_V message.

**Comment by mjcarroll** [comment]
Yes, my understanding was just going off the PR description.  Once I started looking at the implementation, I realized my mistake.  So in coordination with https://github.com/gazebosim/ros_gz/pull/261, I believe that the ROS message parameter entries will be:

```
param_0/model: quadrotor_1
param_0/range: 12.0
param_0/rssi: -45.0
```

And so on for each entry.

File: mbzirc_ign/src/RFRange.cc [hunk 1]
```diff
@@ -0,0 +1,534 @@
...
+    const ignition::gazebo::UpdateInfo &_info,
+    ignition::gazebo::EntityComponentManager &_ecm)
+{
+  IGN_PROFILE("RFRange::PreUpdate");
+
+  _ecm.EachNew<RFRangeType,
+               ignition::gazebo::components::WorldPose>(
+    [&](const ignition::gazebo::Entity &_entity,
+        const RFRangeType *_custom,
+        const ignition::gazebo::components::WorldPose *_pose)->bool
```
Comment (iche033) [inline mbzirc_ign/src/RFRange.cc:430]: `_pose` not used?

**Review by iche033 — COMMENTED** [review]
works for me. Just a couple of minor comments

File: mbzirc_ign/src/RFRange.cc (hunk 1 above)
Comment (caguero) [reply mbzirc_ign/src/RFRange.cc]: Removed in [8450745](https://github.com/osrf/mbzirc/pull/151/commits/8450745fb3e3afdf3bc481d5f44c0bf25c237b37).

File: mbzirc_ign/src/RFRange.cc [hunk 2]
```diff
@@ -0,0 +1,534 @@
...
+
+  _ecm.SetParentEntity(entity, _entity);
+  _ecm.CreateComponent(entity, RFRangeType());
+
+  math::Pose3d pose;
+  if (_ecm.EntityHasComponentType(_entity, gazebo::components::Model::typeId))
+  {
+    auto modelPose = _ecm.Component<gazebo::components::Pose>(_entity);
+    pose += modelPose->Data();
+    _ecm.CreateComponent(entity, gazebo::components::WorldPose(pose));
```
[the remark this answers is not in the document]
Comment (caguero) [reply mbzirc_ign/src/RFRange.cc]: We don't need it. Removed in [8450745](https://github.com/osrf/mbzirc/pull/151/commits/8450745fb3e3afdf3bc481d5f44c0bf25c237b37).

**Comment by caguero** [comment]
> the RF range sensor works for me as advertised. We're looking to provide multiple configurations, e.g. one short range and one long range. Teams must use the same model on all robots. One idea is to let the sensor set one of the RF params that affects its working range when it's loaded, e.g. `tx_power`?. I can work on this. @caguero let me know what params to change and what are the reasonable values to set if we want a short and long range model.

Thanks! Parameters adjusted in your PR directly.

**Review by iche033 — APPROVED** [review]

**Comment by caguero** [comment]
API documentation updated in https://github.com/osrf/mbzirc/wiki/Robot-and-sensor-APIs#uav-and-usv-payload-sensors.

**Pull request merged**